db/sync_test.go

```diff
@@ -206,3 +212,123 @@ func (f *fakePuller) Pull(noms types.ValueReadWriter, baseState Commit, url stri
 	}
 	return Commit{}, f.clientViewInfo, errors.New(f.err)
 }
+
+func TestDB_MaybeEndSync(t *testing.T) {
+	assert := assert.New(t)
+	d := datetime.Now()
+
+	tests := []struct {
```

Comment:
[taste] I'd also add tests for the error cases that are easy to inject. Looking at `EndSync()` it's easy to test for all the various commits not existing and not unmarshaling.